src/pipeline.cpp

```diff
@@ -619,6 +619,37 @@ namespace librealsense
         return false;
     }
 
+    bool pipeline::try_wait_for_frames(frame_holder* frame, unsigned int timeout_ms)
```

Comment:
This seems like duplicate code, can we reuse?

Reply:
Thought about adding try catch for run-time errors instead of duplicating the code, but than I assume that wait_for_frames throws only run-time exceptions in case of timeout, don't want to catch exception of invalid input for example. Is this a reasonable assumption?

Reply:
What about the other way around - `wait_for_frames` calls `try_wait_for_frames` and throws if it returns `false`?

Reply:
In case of device disconnected should I return false or throw exception in try_wait_for frames? I assumed that I should return false, I can't differ between the two cases in which run-time exception is thrown